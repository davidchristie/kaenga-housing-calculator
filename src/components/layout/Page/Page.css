.Page {
  padding-top: 100px;
}
