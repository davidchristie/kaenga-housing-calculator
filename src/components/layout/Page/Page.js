import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { Nav, Navbar, NavItem } from 'react-bootstrap'
import { Link } from 'react-router'
import { LinkContainer } from 'react-router-bootstrap'

import PriceRange from '../../../containers/reports/PriceRange'
import './Page.css'
import Footer from './Footer'
import logo from './logo.png'

class Page extends Component {
  componentWillMount () {
    this.props.loadSuburbs()
  }

  render () {
    return (
      <div className='Page'>
        <Navbar fixedTop>
          <Navbar.Header>
            <Link to='/'>
              <Navbar.Brand>
                <img
                  alt='Kaenga logo'
                  src={logo}
                  style={{height: '80px'}}
                />
              </Navbar.Brand>
            </Link>
            <Navbar.Toggle />
          </Navbar.Header>
          <Navbar.Collapse>
            <Nav bsStyle='tabs' pullRight>
              <LinkContainer to='/report'>
                <NavItem>
                  <span>Price: </span>
                  <PriceRange />
                </NavItem>
              </LinkContainer>
            </Nav>
          </Navbar.Collapse>
        </Navbar>
        {this.props.children}
        <Footer />
      </div>
    )
  }
}

Page.defaultProps = {
  loadSuburbs: () => {}
}

Page.propTypes = {
  loadSuburbs: PropTypes.func.isRequired
}

export default Page
