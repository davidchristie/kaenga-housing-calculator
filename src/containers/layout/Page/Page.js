import { connect } from 'react-redux'

import { loadSuburbs } from '../../../actions/suburbs'
import Page from '../../../components/layout/Page'
import { getReport } from '../../../selectors/report'

export default connect(
  state => {
    return {
      range: getReport(state).range
    }
  },
  dispatch => {
    return {
      loadSuburbs: () => dispatch(loadSuburbs())
    }
  }
)(Page)
