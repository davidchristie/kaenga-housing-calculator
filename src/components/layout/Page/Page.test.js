import React from 'react'
import ReactTestUtils from 'react-dom/test-utils'

import Page from './Page'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<Page />)).toMatchSnapshot()
})
