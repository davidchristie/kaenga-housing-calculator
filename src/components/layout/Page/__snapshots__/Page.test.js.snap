exports[`test matches snapshot 1`] = `
<div
  className="Page">
  <Uncontrolled(Navbar)
    bsStyle="default"
    fixedTop={true}>
    <NavbarHeader>
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/">
        <NavbarBrand>
          <img
            alt="Kaenga logo"
            src="logo.png"
            style={
              Object {
                "height": "80px",
              }
            } />
        </NavbarBrand>
      </Link>
      <NavbarToggle />
    </NavbarHeader>
    <NavbarCollapse>
      <Nav
        bsClass="nav"
        justified={false}
        pullLeft={false}
        pullRight={true}
        stacked={false}>
        <NavItem
          active={false}
          disabled={false}
          style={
            Object {
              "display": "none",
            }
          }>
          <span>
            Price: 
          </span>
          <Connect(PriceRange) />
        </NavItem>
      </Nav>
    </NavbarCollapse>
  </Uncontrolled(Navbar)>
  <Footer />
</div>
`;
